app/src/main/java/com/x8bit/bitwarden/data/platform/repository/ServerConfigRepositoryImpl.kt

```diff
@@ -0,0 +1,60 @@
+package com.x8bit.bitwarden.data.platform.repository
+
+import com.x8bit.bitwarden.data.platform.datasource.disk.ConfigDiskSource
+import com.x8bit.bitwarden.data.platform.datasource.disk.model.ServerConfig
+import com.x8bit.bitwarden.data.platform.datasource.network.service.ConfigService
+import com.x8bit.bitwarden.data.platform.manager.dispatcher.DispatcherManager
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import java.time.Instant
+
+class ServerConfigRepositoryImpl(
+    private val configDiskSource: ConfigDiskSource,
+    private val configService: ConfigService,
+    dispatcherManager: DispatcherManager,
+) : ServerConfigRepository {
+
+    private val unconfinedScope = CoroutineScope(dispatcherManager.unconfined)
+    override suspend fun getServerConfig(forceRefresh: Boolean): ServerConfig? {
+        val localConfig = configDiskSource.serverConfig
+        val needsRefresh = localConfig == null || localConfig.let {
+            Instant.ofEpochMilli(it.lastSync).isAfter(
+                Instant.now().plusSeconds(MINIMUM_CONFIG_SYNC_INTERVAL)
+            )
+        }
+
+        if (needsRefresh || forceRefresh) {
+            configService.getConfig().fold(
+                onSuccess = { configResponse ->
+                    val serverConfig = ServerConfig(
+                        lastSync = Instant.now().toEpochMilli(),
+                        serverData = configResponse
+                    )
+                    configDiskSource.serverConfig = serverConfig
+                    return serverConfig
+                },
+                onFailure = {
+                    throw it
```

Comment:
This seems a bit weird to me? We just crash if the network request fails?

Reply:
Forgot about it. Nope, we should just handle the success and return the state if the request failed.